File: binary_search/javascript/BinarySearch.js
class BinarySearch {
  static binarySearch(arr, key) {

    let l = arr.length;
    let left = 0; let right = l - 1;
    let mid;
    while(left <= right) {
      mid = left + ((right - left) / 2 | 0);
      if(key < arr[mid]) {
        right = mid - 1;
      } else if(key > arr[mid]) {
        left = mid + 1;
      } else {
        return mid;
      }
    }
    return -1;
  }
}

File: binary_search/ruby/binary_search.rb
def binary_search(arr, key)

  l = arr.length
  left = 0; right = l - 1
  while left <= right do
    mid = left + (right - left) / 2
    if key < arr[mid]
      right = mid - 1
    elsif key > arr[mid]
      left = mid + 1
    else
      return mid
    end
  end
  return -1
end
